src/java.base/share/classes/java/io/Writer.java

```diff
@@ -305,33 +275,19 @@ public void write(String str) throws IOException {
      *          If an I/O error occurs
      */
     public void write(String str, int off, int len) throws IOException {
-        Object lock = this.lock;
-        if (lock instanceof InternalLock locker) {
-            locker.lock();
-            try {
-                implWrite(str, off, len);
-            } finally {
-                locker.unlock();
-            }
-        } else {
-            synchronized (lock) {
-                implWrite(str, off, len);
-            }
-        }
-    }
-
-    private void implWrite(String str, int off, int len) throws IOException {
-        char cbuf[];
-        if (len <= WRITE_BUFFER_SIZE) {
-            if (writeBuffer == null) {
-                writeBuffer = new char[WRITE_BUFFER_SIZE];
+        synchronized (lock) {
+            char cbuf[];
```

Comment:
As annoying as this is, but https://bugs.openjdk.org/browse/JDK-8272626 didn't fix c-style arrays in method bodies.  Sad.